spring-security-oauth2/src/main/java/org/springframework/security/oauth2/client/token/OAuth2AccessTokenExpiryChecker.java

```diff
@@ -0,0 +1,45 @@
+package org.springframework.security.oauth2.client.token;
+
+import org.springframework.security.oauth2.common.OAuth2AccessToken;
+
+import java.util.Calendar;
+
+/**
+ * Checks expiration of an OAuth2AccessToken while also taking a clock skew into account.
+ *
+ * @author Andreas Klöber
+ * @deprecated See the <a href="https://github.com/spring-projects/spring-security/wiki/OAuth-2.0-Migration-Guide">OAuth 2.0 Migration Guide</a> for Spring Security 5.
+ */
+@Deprecated
+public class OAuth2AccessTokenExpiryChecker {
```

Comment:
We cannot add public API's going forward as of 2.5.0 as it was the final minor release given this project is coming to EOL - see [announcement](https://spring.io/blog/2020/05/07/end-of-life-for-spring-security-oauth). The changes in this PR need to be patch changes.

Here is a suggestion for the patch change:

- Introduce `private int clockSkew = 30` in `AccessTokenProviderChain` - same as `OAuth2RestTemplate`
- When `OAuth2RestTemplate.setClockSkew()` is called, it will also set `AccessTokenProviderChain.clockSkew` member to the same value using reflection. Take a look at this [test](https://github.com/spring-projects/spring-security-oauth/blob/304c43bed142853442d1a76e5969e337cc887378/spring-security-oauth2/src/test/java/org/springframework/security/oauth2/provider/token/store/jwk/JwkTokenStoreTests.java#L63) to see how you can leverage `ReflectionUtils`.

I know this is not a pretty fix but it will work for the patch.

Reply:
@jgrandja Thanks for the feedback. I only introduced `OAuth2AccessTokenExpiryChecker` class as I did not find a better place to share the expiration-check-logic.  I'll have a look at your suggestion.